src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,115 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import StatusList from '@digitalbazaar/vc-status-list';
+import { URL } from 'url';
+import { ConfigService } from '@nestjs/config';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import {
+  CredentialWithStatusDto,
+  StatusList2021EntryDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  async createEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    const statusListCredential = new URL(
+      `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+      this.configService.get('STATUS_LIST_DOMAIN')
+    );
```

Comment:
I wonder if we should persist the entries we create and check if we have already created an entry with the same `credential.id`. Ideally the client should be creating unique ids for each credential but if they don't I think it could cause a problem where we would be giving out two credentials with the same status list entry and then subsequent updates to the statusListCredential could overwrite eachother.

Reply:
Yeah, maybe this is a good idea. I will add it